crates/typed-store/src/rocks/mod.rs

```diff
@@ -578,6 +709,45 @@ impl DBBatch {
     }
 }
 
+pub struct DBTransaction<'a> {
+    rocksdb: Arc<RocksDB>,
+    transaction: Transaction<'a, rocksdb::OptimisticTransactionDB>,
+}
+
+impl<'a> DBTransaction<'a> {
+    pub fn new(db: &'a Arc<RocksDB>) -> Result<Self, TypedStoreError> {
+        Ok(Self {
+            rocksdb: db.clone(),
+            transaction: db.transaction()?,
+        })
+    }
+
+    pub fn insert_batch<J: Borrow<K>, K: Serialize, U: Borrow<V>, V: Serialize>(
```

Comment:
We will also need a `delete_batch` and `delete_range` for txns

Reply:
`delete_batch`  for sure, but I was thinking to split PR at this point
`delete_range` is not supported for optimistic transaction db